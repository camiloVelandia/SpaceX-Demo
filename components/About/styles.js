import styled from "styled-components";

export const Section = styled.section`
  position:relative;
  margin-top: 90px;
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 700px;
  /* height: 100vh; */
  padding:80px 20px;
  width: 100%;
  background-image: url("./images/Dragon.jpg");
  background-position: top;
  background-size: cover;
  background-repeat: no-repeat;
  &::after {
    content: "";
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: rgba(0, 0, 0, 0.7);
    z-index: 0;
  }
`;
export const Container = styled.div`
  width: 80%;
  background-color: rgba(0, 0, 0, 0.8);
  position:relative;
  z-index:5;
  padding:40px;
  border-radius:25px;
  display:flex;
  justify-content:space-between;
  align-items:center;
  flex-wrap:wrap
`;
export const Article = styled.article`
  margin: 50px 0;
  width: 100%;
  @media (min-width: 824px) {
    width: 45%;
  }
`;
export const Title = styled.h2`
  margin: 0;
  font-size: 35px;
  text-transform: uppercase;
  @media (min-width: 939px) {
  font-size: 40px;
  }
  @media (min-width: 1121px) {
  font-size: 48px;
  }
`;
export const Text = styled.p`
 margin:0;
 font-size:18px;
 max-width:700px;
 padding-top:30px;
`;
export const Grid = styled.div`
max-width:500px;
 display:flex;
 justify-content:space-between;
`;
export const Item = styled.div`
padding-top:20px;
 & h3{
   font-size:18px;
   font-weight:400;
   text-transform:uppercase;
 }
 & ul{
   list-style:none;
   padding:0;

 }
`;
