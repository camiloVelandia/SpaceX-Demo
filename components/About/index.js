import React, {useState, useEffect} from 'react';
import getCompanyInfo from '../../services/getCompanyInfo';
import { Section, Container, Article, Title,Text,Grid,Item } from './styles';

const About = () => {

  const [info, setinfo] = useState({links:{}} )


  useEffect(()=>{
    getCompanyInfo()
    .then(data=>setinfo({...data}))
  },[])

  return (
    <Section>
      <Container>
        <Article>
          <Title>about: {info.name}</Title>
          <Text>{info.summary}</Text>
          <Grid>
            <Item>
              <h3>company info</h3>
              <ul>
                <li>Founded: {info.founded}</li>
                <li>CEO: {info.ceo}</li>
                <li>CTO: {info.cto}</li>
                <li>COO: {info.coo}</li>
                <li>CTO propulsion: {info.cto_propulsion}</li>
              </ul>
            </Item>
            <Item>
              <h3>More</h3>
              <ul>
                <li>Employees: {info.employees}</li>
                <li>Launch Sites: {info.launch_sites}</li>
                <li>Valuation: {info.valuation}</li>
                <li>
                  Website:{" "}
                  <a href={info.links.website} target="_blank">
                    {info.links.website}
                  </a>
                </li>
                <li>
                  twitter:{" "}
                  <a href={info.links.twitter} target="_blank">
                    {info.links.twitter}
                  </a>
                </li>
              </ul>
            </Item>
          </Grid>
        </Article>
        <Article>
          <Title>about: me</Title>
          <Text>
            Hi, I'm Camilo Velandia, a passionate self-taught frontEnd web
            developer from Colombia. with an interest in web and mobile
            projects, with user-centered approaches. currently part of the
            Platzi master program.
          </Text>
          <Grid>
            <Item>
              <h3>experience</h3>
              <ul>
                <li>Software Developer at Platzi master</li>
                <li>Frontend Developer at 360 PMI</li>
                <li>
                  <a
                    href="https://www.linkedin.com/in/camilo-velandia/"
                    target="_blank"
                  >
                    Find me in Linkedin{" "}
                  </a>
                </li>
                <li>
                  <a href="https://twitter.com/xander_coder" target="_blank">
                    Find me in Twitter
                  </a>
                </li>
                <li>
                  <a href="https://github.com/camiloVelandia" target="_blank">
                    Find me in Github
                  </a>
                </li>
              </ul>
            </Item>
            <Item>
              <h3> tools</h3>
              <ul>
                <li>Javascript</li>
                <li>React js</li>
                <li>Redux</li>
                <li>Next js</li>
                <li>Styled components</li>
                <li>Git</li>
              </ul>
            </Item>
          </Grid>
        </Article>
      </Container>
    </Section>
  );
};

export default About;