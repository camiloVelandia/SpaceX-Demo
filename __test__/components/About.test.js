import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import About from "../../components/About";

describe("<About/>", () => {
  const about = mount(<About />);
  test("render del componente about", () => {
    expect(about.length).toEqual(1);
  });
});

describe("about snaaapshot", () => {
  test("Comprobar ui del about", () => {
    const about = create(<About />);
    expect(about.toJSON()).toMatchSnapshot();
  });
});
