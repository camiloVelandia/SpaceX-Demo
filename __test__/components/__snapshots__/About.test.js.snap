// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about snaaapshot Comprobar ui del about 1`] = `
<section
  className="styles__Section-egwdpf-0 fdTLyL"
>
  <div
    className="styles__Container-egwdpf-1 kGKabv"
  >
    <article
      className="styles__Article-egwdpf-2 jTrjbM"
    >
      <h2
        className="styles__Title-egwdpf-3 duaghV"
      >
        about: 
      </h2>
      <p
        className="styles__Text-egwdpf-4 HaXMz"
      />
      <div
        className="styles__Grid-egwdpf-5 dFiVAz"
      >
        <div
          className="styles__Item-egwdpf-6 fNlEqf"
        >
          <h3>
            company info
          </h3>
          <ul>
            <li>
              Founded: 
            </li>
            <li>
              CEO: 
            </li>
            <li>
              CTO: 
            </li>
            <li>
              COO: 
            </li>
            <li>
              CTO propulsion: 
            </li>
          </ul>
        </div>
        <div
          className="styles__Item-egwdpf-6 fNlEqf"
        >
          <h3>
            More
          </h3>
          <ul>
            <li>
              Employees: 
            </li>
            <li>
              Launch Sites: 
            </li>
            <li>
              Valuation: 
            </li>
            <li>
              Website:
               
              <a
                target="_blank"
              />
            </li>
            <li>
              twitter:
               
              <a
                target="_blank"
              />
            </li>
          </ul>
        </div>
      </div>
    </article>
    <article
      className="styles__Article-egwdpf-2 jTrjbM"
    >
      <h2
        className="styles__Title-egwdpf-3 duaghV"
      >
        about: me
      </h2>
      <p
        className="styles__Text-egwdpf-4 HaXMz"
      >
        Hi, I'm Camilo Velandia, a passionate self-taught frontEnd web developer from Colombia. with an interest in web and mobile projects, with user-centered approaches. currently part of the Platzi master program.
      </p>
      <div
        className="styles__Grid-egwdpf-5 dFiVAz"
      >
        <div
          className="styles__Item-egwdpf-6 fNlEqf"
        >
          <h3>
            experience
          </h3>
          <ul>
            <li>
              Software Developer at Platzi master
            </li>
            <li>
              Frontend Developer at 360 PMI
            </li>
            <li>
              <a
                href="https://www.linkedin.com/in/camilo-velandia/"
                target="_blank"
              >
                Find me in Linkedin
                 
              </a>
            </li>
            <li>
              <a
                href="https://twitter.com/xander_coder"
                target="_blank"
              >
                Find me in Twitter
              </a>
            </li>
            <li>
              <a
                href="https://github.com/camiloVelandia"
                target="_blank"
              >
                Find me in Github
              </a>
            </li>
          </ul>
        </div>
        <div
          className="styles__Item-egwdpf-6 fNlEqf"
        >
          <h3>
             tools
          </h3>
          <ul>
            <li>
              Javascript
            </li>
            <li>
              React js
            </li>
            <li>
              Redux
            </li>
            <li>
              Next js
            </li>
            <li>
              Styled components
            </li>
            <li>
              Git
            </li>
          </ul>
        </div>
      </div>
    </article>
  </div>
</section>
`;
